<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>913</width>
    <height>507</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="table_caption_lb">
          <property name="text">
           <string>Table caption</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLineEdit" name="search_product_le">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="placeholderText">
           <string>Search product...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="clear_product_search_pb">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableWidget" name="products_tw">
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="showGrid">
         <bool>false</bool>
        </property>
        <attribute name="horizontalHeaderHighlightSections">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderHighlightSections">
         <bool>true</bool>
        </attribute>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QToolButton" name="home_pb">
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="pg_up_pb">
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="pg_down_pb">
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="end_pb">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="detailview_ckb">
          <property name="text">
           <string>Show &amp;details</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="categories_dkw">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Categories</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLineEdit" name="search_category_le">
           <property name="placeholderText">
            <string>Search category...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="clear_category_search_pb">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QListWidget" name="categories_lw">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="cartitems_dkw">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Cart Items</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_9">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QTableWidget" name="cartitems_tw">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderHighlightSections">
          <bool>true</bool>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>913</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_view">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_toggle_categories"/>
    <addaction name="action_toggle_cartitems"/>
    <addaction name="separator"/>
    <addaction name="action_show_sales"/>
   </widget>
   <widget class="QMenu" name="menu_file">
    <property name="geometry">
     <rect>
      <x>298</x>
      <y>156</y>
      <width>144</width>
      <height>277</height>
     </rect>
    </property>
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_checkout"/>
    <addaction name="action_clear_cart"/>
    <addaction name="separator"/>
    <addaction name="action_add_product"/>
    <addaction name="action_add_category"/>
    <addaction name="separator"/>
    <addaction name="action_preferences"/>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_about"/>
    <addaction name="action_about_qt"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_view"/>
   <addaction name="menu_help"/>
  </widget>
  <action name="action_add_product">
   <property name="text">
    <string>Add Product</string>
   </property>
   <property name="statusTip">
    <string>Add a new product</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_add_category">
   <property name="text">
    <string>Add Category</string>
   </property>
   <property name="toolTip">
    <string>Add a new category</string>
   </property>
   <property name="statusTip">
    <string>Add a new product category</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+N</string>
   </property>
  </action>
  <action name="action_add_user">
   <property name="text">
    <string>Create User</string>
   </property>
   <property name="statusTip">
    <string>Create a new user account</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="statusTip">
    <string>Exit application</string>
   </property>
   <property name="whatsThis">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="action_preferences">
   <property name="text">
    <string>Preferences</string>
   </property>
   <property name="statusTip">
    <string>Preferences</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+J</string>
   </property>
  </action>
  <action name="action_show_sales">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show sales</string>
   </property>
   <property name="statusTip">
    <string>Show sales window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="action_checkout">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Checkout</string>
   </property>
   <property name="statusTip">
    <string>Go to checkout</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Return</string>
   </property>
  </action>
  <action name="action_clear_cart">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Clear Cart Items</string>
   </property>
   <property name="statusTip">
    <string>Remove all items from the cart</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+-</string>
   </property>
  </action>
  <action name="action_about_qt">
   <property name="text">
    <string>About Qt</string>
   </property>
   <property name="statusTip">
    <string>About Qt</string>
   </property>
  </action>
  <action name="action_toggle_categories">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Categories</string>
   </property>
   <property name="statusTip">
    <string>Show/Hide Categories</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="action_toggle_cartitems">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>CartItems</string>
   </property>
   <property name="statusTip">
    <string>Show/Hide CartItems</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>search_category_le</tabstop>
  <tabstop>clear_category_search_pb</tabstop>
  <tabstop>categories_lw</tabstop>
  <tabstop>cartitems_tw</tabstop>
  <tabstop>search_product_le</tabstop>
  <tabstop>clear_product_search_pb</tabstop>
  <tabstop>home_pb</tabstop>
  <tabstop>pg_up_pb</tabstop>
  <tabstop>pg_down_pb</tabstop>
  <tabstop>detailview_ckb</tabstop>
  <tabstop>products_tw</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>action_toggle_categories</sender>
   <signal>triggered(bool)</signal>
   <receiver>categories_dkw</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_toggle_cartitems</sender>
   <signal>triggered(bool)</signal>
   <receiver>cartitems_dkw</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_category_search_pb</sender>
   <signal>clicked()</signal>
   <receiver>search_category_le</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>241</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_category_search_pb</sender>
   <signal>clicked()</signal>
   <receiver>search_category_le</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>180</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_product_search_pb</sender>
   <signal>clicked()</signal>
   <receiver>search_product_le</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>883</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>834</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_product_search_pb</sender>
   <signal>clicked()</signal>
   <receiver>search_product_le</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>885</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>830</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>home_pb</sender>
   <signal>clicked()</signal>
   <receiver>products_tw</receiver>
   <slot>scrollToTop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>307</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel">
     <x>331</x>
     <y>390</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>end_pb</sender>
   <signal>clicked()</signal>
   <receiver>products_tw</receiver>
   <slot>scrollToBottom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>421</x>
     <y>455</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
